lib/msf/core/db_manager/vuln.rb

```diff
@@ -46,7 +46,8 @@ def find_vuln_by_details(details_map, host, service=nil)
 
   def find_vuln_by_refs(refs, host, service=nil)
     ref_ids = refs.find_all { |ref| ref.name.starts_with? 'CVE-'}
-    host.vulns.includes(:refs).where(service_id: service.try(:id), refs: { id: ref_ids}).first
+    relation = host.vulns.includes(:refs)
+    relation.where(service_id: service.try(:id), refs: { id: ref_ids}).first || relation.where(refs: { id: ref_ids}).first
```

Comment:
I would really prefer to break this into multiple lines. Maybe something like this:
```ruby
# First try with the given service (or a NULL if we don't have one)
vuln = relation.where(service_id: service.try(:id), refs: { id: ref_ids}).first
# If that didn't return one, try again for any service matching these refs
vuln ||= relation.where(refs: { id: ref_ids}).first
vuln
```